src/main/scala/org/camunda/feel/impl/parser/FeelParser.scala

```diff
@@ -43,92 +56,133 @@ object FeelParser extends JavaTokenParsers {
                "for",
                "between",
                "instance",
-               "of").!)
+               "of"))
 
   // list of built-in function names with whitespaces
   // -- other names match the 'function name' pattern
-  private def builtinFunctionName[_: P]: P[String] = P[String](
-    StringIn(
-      "date and time",
-      "years and months duration",
-      "string length",
-      "upper case",
-      "lower case",
-      "substring before",
-      "substring after",
-      "starts with",
-      "ends with",
-      "list contains",
-      "insert before",
-      "index of",
-      "distinct values",
-      "get entries",
-      "get value",
-      "is defined",
-      "day of year",
-      "day of week",
-      "month of year",
-      "week of year"
-    ).!
-  )
+  private def builtinFunctionName[_: P] =
+    P(
+      StringIn(
+        "date and time",
+        "years and months duration",
+        "string length",
+        "upper case",
+        "lower case",
+        "substring before",
+        "substring after",
+        "starts with",
+        "ends with",
+        "list contains",
+        "insert before",
+        "index of",
+        "distinct values",
+        "get entries",
+        "get value",
+        "is defined",
+        "day of year",
+        "day of week",
+        "month of year",
+        "week of year"
+      )
+    )
 
   // list of built-in function parameter names with whitespaces
   // -- other names match the 'parameter name' pattern
-  private def builtinFunctionParameterNames[_: P]: P[String] =
-    P[String](
-      StringIn("start position", "grouping separator", "decimal separator").!)
+  private def builtinFunctionParameterNames[_: P] =
+    P(StringIn("start position", "grouping separator", "decimal separator"))
 
-  private def identifier[_: P]: P[String] = P(!(reservedWord ~~ !ident) ~ ident)
+  // an identifier which is not a reserved word. however, it can contain a reserved word.
+  private def identifier[_: P] = P(reservedWord.? ~~ javaLikeIdentifier)
```

Comment:
The reservedWord is no longer repeated. What if I write `foobarX` (where both foo and bar are reserved), is this then failing? Also, how does this allow for something like `prefixfooX` (where foo is reserved)?

Reply:
The parser accepts two reserved words or a reserved word with a prefix as a valid name. Only reserved words are not allowed. 

If it parses `reservedWord` then it must be followed by `javaLikeIdentifier` (any additional char). 
If it has a prefix then it is not parsed as `reservedWord`.